util/annotations/helpers.go

```diff
@@ -87,3 +92,27 @@ func hasAnnotation(o metav1.Object, annotation string) bool {
 	_, ok := annotations[annotation]
 	return ok
 }
+
+// hasAnnotationValue returns true if the object has an annotation with the specified value.
+func hasAnnotationValue(o metav1.Object, annotation, value string) bool {
+	annotations := o.GetAnnotations()
+	if annotations == nil {
+		return false
+	}
+	if val, ok := annotations[annotation]; ok {
+		return val == value
+	}
+	return false
+}
+
+// hasTruthyAnnotation evaluates truthiness permissively, only return false if
+// 1) the annotation isn't present at all
+// 2) the annotation is explicitly set to "false".
+// All other value permutations are ignored and considered valid.
+// tl;dr We're looking for boolean-type annotations but respecting that some folks might want to explicitly indicate "false".
+func hasTruthyAnnotation(o metav1.Object, annotation string) bool {
```

Comment:
this function does not seem used currently since `ReplicasManagedByExternalAutoscaler` is using `hasAnnotationValue`. Not sure if intended or not?

Reply:
I left this in place in case review feedback suggested we wanted to go back to using a "truthy" bool annotation instead of one that expects a well-known value. It appears we have consensus that we want the well-known value option, so I'll remove this!